dotnet/samples/GettingStartedWithProcesses/Step04/Steps/RenderMessageStep.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Diagnostics;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Step04.Steps;
+
+/// <summary>
+/// Displays output to the user.  While in this case it is just writing to the console,
+/// in a real-world scenario this would be a more sophisticated rendering system.  Isolating this
+/// rendering logic from the internal logic of other process steps simplifies responsibility contract
+/// and simplifies testing and state management.
+/// </summary>
+public class RenderMessageStep : KernelProcessStep
+{
+    public static class Functions
+    {
+        public const string RenderDone = nameof(RenderMessageStep.RenderDone);
+        public const string RenderError = nameof(RenderMessageStep.RenderError);
+        public const string RenderInnerMessage = nameof(RenderMessageStep.RenderInnerMessage);
+        public const string RenderMessage = nameof(RenderMessageStep.RenderMessage);
+        public const string RenderUserText = nameof(RenderMessageStep.RenderUserText);
+    }
+
+    private readonly static Stopwatch s_timer = Stopwatch.StartNew();
+
+    /// <summary>
+    /// Render an explicit message to indicate the process has completed in the expected state.
+    /// </summary>
+    /// <remarks>
+    /// If this message isn't rendered, the process is considered to have failed.
+    /// </remarks>
+    [KernelFunction]
+    public void RenderDone()
+    {
+        Render("DONE!");
+    }
+
+    /// <summary>
+    /// Render exception
+    /// </summary>
+    [KernelFunction]
+    public void RenderError(Exception exception)
+    {
+        string message = string.IsNullOrWhiteSpace(exception.Message) ? "Unexpected failure" : exception.Message;
```

Comment:
var?

Reply:
I'm not a fan of implicit typing for readability / discovery aspects...particularly as it relates to long-term code maintenance.